Jenkinsfile

```
@@ -16,6 +16,64 @@ def parentZoneResourceGroup = 'jenkins-bkpr-rg'
 // Force using our pod
 def label = env.BUILD_TAG.replaceAll(/[^a-zA-Z0-9-]/, '-').toLowerCase()
 
+def scmCheckout() {
+    def repo_url = env.GITHUB_REPO_GIT_URL
+    def sha = env.GITHUB_PR_HEAD_SHA ? env.GITHUB_PR_HEAD_SHA : env.GITHUB_BRANCH_HEAD_SHA
```

Fwiw, groovy has an "elvis operator", for exactly this "fallback default" situation (should be equivalent to your long version):

```suggestion
    def sha = env.GITHUB_PR_HEAD_SHA ?: env.GITHUB_BRANCH_HEAD_SHA
```

---

nice! will update